ui/src/adapters/parsers/forms.ts

```diff
@@ -143,33 +206,13 @@ const parseIssueCredentialFormDataAttributes = (
     },
     { data: [], success: true }
   );
-};
+}
 
-export const parseLinkExpirationDate = StrictSchema<{
-  linkExpirationDate: dayjs.Dayjs | null;
-}>()(
-  z.object({
-    linkExpirationDate: dayjsInstance.nullable(),
-  })
-);
+// Exports
 
-const issueCredentialFormDataInput = z.object({
-  attributes: z.object({
-    attributes: z.record(z.unknown()),
-    expirationDate: dayjsInstance.nullish(),
-  }),
-  issuanceMethod: z.object({
-    linkExpirationDate: dayjsInstance.nullish(),
-    linkExpirationTime: dayjsInstance.nullish(),
-    linkMaximumIssuance: z.number().optional(),
-  }),
-});
-
-type IssueCredentialFormDataInput = z.infer<typeof issueCredentialFormDataInput>;
-
-export const issueCredentialFormData = (schemaAttributes: SchemaAttribute[]) => {
-  return StrictSchema<IssueCredentialFormDataInput, CredentialForm>()(
-    issueCredentialFormDataInput.transform(
+export function getIssueCredentialFormData(schemaAttributes: SchemaAttribute[]) {
```

Comment:
This is indeed a parser.

I'd remove the parser `issueCredentialFormDataInputParser` above and integrate it here along with the transformer:

```typescript
export function issueCredentialFormDataParser(schemaAttributes: SchemaAttribute[]) {
  return getStrictParser<IssueCredentialFormDataInput, CredentialForm>()(
    z
      .object({
        attributes: z.object({
          attributes: z.record(z.unknown()),
          expirationDate: dayjsInstanceParser.nullable().optional(),
        }),
        issuanceMethod: z.object({
          linkExpirationDate: dayjsInstanceParser.nullable().optional(),
          linkExpirationTime: dayjsInstanceParser.nullable().optional(),
          linkMaximumIssuance: z.number().optional(),
        }),
      })
      .transform(
        (
          {
            attributes: { attributes, expirationDate },
            issuanceMethod: { linkExpirationDate, linkExpirationTime, linkMaximumIssuance },
          },
          zodRefinementCtx
        ) => {
          const attributesParsingResult = parseIssueCredentialFormDataAttributes(
            attributes,
            schemaAttributes
          );

          if (attributesParsingResult.success) {
            const linkAccessibleUntil = buildLinkAccessibleUntil({
              linkExpirationDate,
              linkExpirationTime,
            });

            const now = new Date().getTime();

            if (linkAccessibleUntil && linkAccessibleUntil.getTime() < now) {
              zodRefinementCtx.addIssue({
                code: z.ZodIssueCode.custom,
                fatal: true,
                message: `${ACCESSIBLE_UNTIL} must be a date/time in the future.`,
              });

              return z.NEVER;
            }

            return {
              attributes: attributesParsingResult.data,
              expiration: expirationDate ? expirationDate.toDate() : undefined,
              linkAccessibleUntil,
              linkMaximumIssuance,
            };
          } else {
            attributesParsingResult.error.issues.forEach(zodRefinementCtx.addIssue);

            return z.NEVER;
          }
        }
      )
  );
}
```